include/picongpu/plugins/transitionRadiation/Particle.hpp

```diff
@@ -0,0 +1,184 @@
+/* Copyright 2013-2019 Heiko Burau, Rene Widera, Richard Pausch, Finn-Ole Carstens
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+namespace picongpu
+{
+namespace plugins
+{
+namespace transitionRadiation
+{
+    /** Particle class for transition radiation calculation.
+     *
+     * @param locationSet global position of the macro-particle
+     * @param momentumSet momentum of macro-particle
+     * @param charge
+     */
+    class Particle
+    {
+    private:
+        float3_X const & momentum;
+        float_X const mass;
+        float3_X location;
+        float_X gamma;
+        float3_X beta;
+        float_X betaAbs;
+
+    public:
+        HDINLINE
+        Particle(
+            float3_X const & locationSet,
+            float3_X const & momentumSet,
+            float_X const massSet
+        ) :
+            location( locationSet ),
+            momentum( momentumSet ),
+            mass( massSet )
+        {
+            gamma = calcGamma( );
+            beta = calcBeta( );
+            betaAbs = math::sqrt( ( beta * beta ).sumOfComponents( ) );
+        }
+
+        //! @return momentum
+        HDINLINE
+        float3_X
+        getMomentum( ) const
+        {
+            return momentum;
+        }
+
+        //! @return normalized momentum
+        HDINLINE
+        float_X
+        getU( ) const
+        {
+            return gamma * betaAbs;
+        }
+
+        //! @return velocity v = beta * c
+        HDINLINE
+        float_X
+        getVel( ) const
+        {
+            return betaAbs * picongpu::SPEED_OF_LIGHT;
+        }
+
+        //! propagates the current window to the foil position
+        HDINLINE
+        void
+        propagate( const float_X & propagationDistance )
+        {
+            location += propagationDistance * beta;
```

Comment:
This is only valid if velocities ae normalized to c and the `propagationDistance` is a time. 
@finnolec is the later the case?